components/brave_sync/client/bookmark_change_processor.cc

```diff
@@ -447,34 +527,265 @@ void BookmarkChangeProcessor::ApplyChangesFromSyncModel(
         }
       }
     } else if (sync_record->action == jslib::SyncRecord::Action::A_CREATE) {
+      bool folder_was_created = false;
+      const bookmarks::BookmarkNode* parent_node = nullptr;
       if (!node) {
         // TODO(bridiver) make sure there isn't an existing record for objectId
-        const bookmarks::BookmarkNode* parent_node =
-            FindParent(bookmark_model_, bookmark_record);
+        parent_node =
+            FindParent(bookmark_model_, bookmark_record, GetPendingNodeRoot());
 
         const BookmarkNode* bookmark_bar = bookmark_model_->bookmark_bar_node();
         bool bookmark_bar_was_empty = bookmark_bar->empty();
+
+        // Use manual, not bookmark_model_->AddXXX, add node/folder to avoid
```

Comment:
this won't be necessary, observer has already paused within the scope of `ApplyChangesFromSyncModel`

Reply:
Here the point is not about our observers, but other observers.
Otherwise, I get the node `Pending nodes` visible on `brave://bookmarks` for the amount of time when it has children. Here I think, observer of JS bookmarks API is triggered. So disabled that observers in that way.

Reply:
I see. Then "Deleted Bookmarks" has the same problem not just "Pending Bookmarks"
What about making both "Deleted Bookmarks" and "Pending Bookmarks" `node->set_visible(false);` again after `UpdateNode` and in the end of `BookmarkNodeRemoved` because it must be set to visible again when observer notified.
Current implementation contains too many copy logic from upstream BookmarkModel and it will be hard to maintain and we have to manually send `BookmarkNodeAdded` for the bookmark to be visible after move out.